mule-apikit-maven-plugin/src/test/java/org/mule/tools/apikit/CreateMojoTest.java

```diff
@@ -21,211 +21,229 @@
+    private File src;
+    private File main;
+    private File app;
+    private File resources;
+    private File api;
+    private File lala;
+    private File project;
+    private File apiFile;
+
+    @Before
+    public void setUp() throws Exception {
+        mojo = new CreateMojo();
+
+
+        project = folder.newFolder("my-project");
+        src = new File(project, "src");
+        main = new File(src, "main");
+        app = new File(main, "mule");
+        resources = new File(main, "resources");
+        api = new File(resources, "api");
+        lala = new File(api, "lala");
+
+
+        api.mkdirs();
+        app.mkdirs();
+        lala.mkdirs();
+
+        // Do
+        apiFile = new File(api, "hello.raml");
+        apiFile.createNewFile();
+        new File(api, "bye.yml").createNewFile();
+        new File(lala, "wow.raml").createNewFile();
+
+        // Don't
+        new File(main, "dont-read.raml").createNewFile();
+
+        // TODO mock properties like this:
+        setVariableValueToObject(mojo, "buildContext", new DefaultBuildContext());
+        setVariableValueToObject(mojo, "log", mock(Log.class));
+        setVariableValueToObject(mojo, "scaffoldingConfigurationFile", new File(this.getClass().getClassLoader().getResource("configuration/full-configuration.json").toURI()));
+    }
+
+    @Test
+    public void testReadScaffoldingConfigurationMojoEmptyConfiguration() throws URISyntaxException, IllegalAccessException {
+        setVariableValueToObject(mojo, "scaffoldingConfigurationFile", new File(this.getClass().getClassLoader().getResource("configuration/empty-configuration.json").toURI()));
+        ScaffoldingConfigurationMojo scaffoldingConfigurationMojo = mojo.readScaffoldingConfigurationMojo();
+        assertTrue(scaffoldingConfigurationMojo.isShowConsole());
+        assertEquals(scaffoldingConfigurationMojo.getExternalCommonFile(), null);
+        assertEquals(scaffoldingConfigurationMojo.getApiId(), null);
+    }
+
+    @Test
+    public void testReadScaffoldingConfigurationMojoFullConfiguration() throws URISyntaxException, IllegalAccessException {
+        setVariableValueToObject(mojo, "scaffoldingConfigurationFile", new File(this.getClass().getClassLoader().getResource("configuration/full-configuration.json").toURI()));
+        ScaffoldingConfigurationMojo scaffoldingConfigurationMojo = mojo.readScaffoldingConfigurationMojo();
+        assertTrue(scaffoldingConfigurationMojo.isShowConsole());
+        assertEquals(scaffoldingConfigurationMojo.getExternalCommonFile(), "globals.xml");
+        assertEquals(scaffoldingConfigurationMojo.getApiId(), "16356937");
+    }
+
+    @Test
+    public void testGetIncludedFiles() throws Exception {
+        List<String> files =
+                mojo.getIncludedFiles(project, new String[]{"src/main/resources/api/**/*.raml", "src/main/resources/api/**/*.yml"},
+                        new String[]{});
+        HashSet<String> set = new HashSet<String>(files);
```

Comment:
You can remove the generic type for diamond in the new statement. I would put a more semantic name and define the variable with the interface rather than the implementation.

Reply:
I haven't edited the whole file as shown, possibly a mvn formatter would have caused it to be marked as edited. Anyway I'll remove the generics and put a better name.

For reference: I have added a line (83), and then added scenarios between 86 and 102.